controllers/datasource_controller.go

```diff
@@ -152,17 +154,30 @@ func (r *DatasourceReconciler) Checkdatasource(ctx context.Context, logger logr.
 	var info any
 	switch instance.Spec.Type() {
 	case arcadiav1alpha1.DatasourceTypeLocal:
-		// FIXME: implement local datasource check when system datasource defined by https://github.com/kubeagi/arcadia/issues/156
-		// 1. read system datasource endpoint
-		// 2. check against pre-denfined rules for local datasource rules
-		return r.UpdateStatus(ctx, instance, nil)
+		var endpoint *arcadiav1alpha1.Endpoint
+		if instance.Spec.Enpoint == nil {
+			system, err := config.GetSystemDatasource(ctx, r.Client)
+			if err != nil {
+				return r.UpdateStatus(ctx, instance, err)
+			}
+			endpoint = system.Spec.Enpoint.DeepCopy()
+			info = system.Spec.OSS.DeepCopy()
```

Comment:
For `local` , the oss info should be 
```yaml
 oss:
   bucket: instance.namespace
```

Reply:
If not exists.we need to create one.

Reply:
the original oss.bucket in system datasource is meant to make `system datasource` be a public `Datasource` (pre-uploaded training data)for other users. 

- System datasource 
    - bucket: {arcadia_namespce}